### ISSUE TITLE
Enunciate 2: Maven has stack overflow while generating docs with v2.0 and Java 7

### ISSUE THREAD

**Issue by raspberrypuppies** [stoicflame/enunciate#197]
I have a webservice that's split into three projects, a parent project that just has a POM file, a "DTO" project that contains POJO's that go in/out of the database, and a webservice project that I am trying to run Enunciate on.

Here's the plugin definition from the pom file: (enunciate.version is 2.0.0)
&lt;plugin&gt;
	&lt;groupId&gt;com.webcohesion.enunciate&lt;/groupId&gt;
	&lt;artifactId&gt;enunciate-maven-plugin&lt;/artifactId&gt;
	&lt;version&gt;${enunciate.version}&lt;/version&gt;
	&lt;executions&gt;
		&lt;execution&gt;
			&lt;goals&gt;
				&lt;goal&gt;assemble&lt;/goal&gt;
			&lt;/goals&gt;
		&lt;/execution&gt;
	&lt;/executions&gt;
&lt;/plugin&gt;

Here's maven's output:

```
C:\Workspaces\BIDMC\bidmc-web-service>mvn clean package
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building bidmc-webservice 1.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading: http://hudson.ma.aptima.com:8081/nexus/content/repositories/Releases/ch/qos/logback/logback-classic/maven-metadata.xml
Downloading: http://hudson.ma.aptima.com:8081/nexus/content/repositories/thirdparty/ch/qos/logback/logback-classic/maven-metadata.xml
Downloading: http://hudson.ma.aptima.com:8081/nexus/content/repositories/snapshots/ch/qos/logback/logback-classic/maven-metadata.xml
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ bidmc-webservice ---
[INFO] Deleting C:\Workspaces\BIDMC\bidmc-web-service\target
[INFO]
[INFO] --- enunciate-maven-plugin:2.0.0:assemble (default) @ bidmc-webservice ---
[INFO] [ENUNCIATE] Using enunciate configuration at C:\Workspaces\BIDMC\bidmc-web-service\enunciate.xml
[INFO] Reflections took 1247 ms to scan 53 urls, producing 145 keys and 145 values
Exception in thread "pool-2-thread-1" java.lang.StackOverflowError
        at java.lang.reflect.Proxy$Key1.<init>(Proxy.java:440)
        at java.lang.reflect.Proxy$KeyFactory.apply(Proxy.java:545)
        at java.lang.reflect.Proxy$KeyFactory.apply(Proxy.java:539)
        at java.lang.reflect.WeakCache.get(WeakCache.java:120)
        at java.lang.reflect.Proxy.getProxyClass0(Proxy.java:419)
        at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:719)
        at sun.reflect.annotation.AnnotationParser$1.run(AnnotationParser.java:305)
        at sun.reflect.annotation.AnnotationParser$1.run(AnnotationParser.java:303)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.reflect.annotation.AnnotationParser.annotationForMap(AnnotationParser.java:303)
        at com.sun.tools.javac.model.AnnotationProxyMaker.generateAnnotation(AnnotationProxyMaker.java:87)
        at com.sun.tools.javac.model.AnnotationProxyMaker.generateAnnotation(AnnotationProxyMaker.java:79)
        at com.sun.tools.javac.code.AnnoConstruct.getAnnotation(AnnoConstruct.java:179)
        at com.webcohesion.enunciate.javac.decorations.element.DecoratedElement.getAnnotation(DecoratedElement.java:256)
        at com.webcohesion.enunciate.javac.decorations.element.DecoratedElement.getAnnotation(DecoratedElement.java:256)
        at com.webcohesion.enunciate.modules.jackson.model.types.JsonTypeFactory.findSpecifiedType(JsonTypeFactory.java:80)
        at com.webcohesion.enunciate.modules.jackson.model.Member.getAccessorType(Member.java:242)
        at com.webcohesion.enunciate.modules.jackson.model.types.JsonTypeFactory.findSpecifiedType(JsonTypeFactory.java:122)
        at com.webcohesion.enunciate.modules.jackson.model.Member.getAccessorType(Member.java:242)
        at com.webcohesion.enunciate.modules.jackson.model.types.JsonTypeFactory.findSpecifiedType(JsonTypeFactory.java:122)
        at com.webcohesion.enunciate.modules.jackson.model.Member.getAccessorType(Member.java:242)
        at com.webcohesion.enunciate.modules.jackson.model.types.JsonTypeFactory.findSpecifiedType(JsonTypeFactory.java:122)
        at com.webcohesion.enunciate.modules.jackson.model.Member.getAccessorType(Member.java:242)
        at com.webcohesion.enunciate.modules.jackson.model.types.JsonTypeFactory.findSpecifiedType(JsonTypeFactory.java:122)
```

**Comment by stoicflame** [comment]
Thanks for the report.

**Comment by stoicflame** [comment]
Fixed at 08163df.

**Comment by stoicflame** [comment]
[Enunciate 2.1 has been released.](https://github.com/stoicflame/enunciate/releases/tag/v2.1.0)
